### PR TITLE
fixed various issues and updating to beta-2

### PR DESCRIPTION
**Pull request by fudgepop01**
this will be updating the design and adding features to beta version 2.
Features include two types of filters and an updated look.

### CODE REVIEW

File: extensions/applicationinsights-debugplugin-js/sample/public/sample.js [hunk 1]
```diff
@@ -104,6 +113,16 @@ function myFunc() {
...
   }
 
+  var container = document.createElement('div');
+  container.style.position = 'absolute';
+  container.style.margin = 'auto';
+  container.style.top = '0';
+  container.style.left = '0';
+  container.style.bottom = '0';
+  container.style.right = '0';
+  container.style.width = "300px";
```
Comment (markwolff) [inline extensions/applicationinsights-debugplugin-js/sample/public/sample.js:123]: nit: consistent quotes

File: extensions/applicationinsights-debugplugin-js/sample/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+presentation/
```
Comment (markwolff) [inline extensions/applicationinsights-debugplugin-js/sample/.gitignore:1]: nit: nl

File: extensions/applicationinsights-debugplugin-js/src/DebugPlugin.ts [hunk 3]
```diff
@@ -152,56 +152,35 @@ export default class DebugPlugin extends BaseTelemetryPlugin {
...
-            [propertiesExt, propertiesProtoFns],
-            [ajaxDependencyExt, ajaxProtoFns],
-            [channelPluginExt, channelProtoFns],
-            [diagLog, diagLogProtoFns]
-          ] as any[]) {
-            for (const key of CoreUtils.objKeys(ext['__proto__'])) {
-              if (key.substring(0, 1) === '_') { continue; }
-              if (CoreUtils.isFunction(ext[key])) {
-                protoFns.push(key);
+          for (let i = 0; i < trackers.length; i++) {
```
Comment (markwolff) [inline extensions/applicationinsights-debugplugin-js/src/DebugPlugin.ts:159]: Can `for ... of` be used here?

File: extensions/applicationinsights-debugplugin-js/src/DebugPlugin.ts [hunk 4]
```diff
@@ -152,56 +152,35 @@ export default class DebugPlugin extends BaseTelemetryPlugin {
...
-              rsp: _self.postProcessItem(tracker) as any as () => InstrumentorHooksCallback
-            });
-
-            debugBins[tracker] = new DebugBin(tracker, 0, debugBinParent, (i + 1) * 50);
           }
 
           // 6. append permanent style
           permStyleEl = document.createElement("style");
+          console.log(permStyle(prefix));
+          console.log(permStyle(prefix).substring(1000));
```
Comment (markwolff) [inline extensions/applicationinsights-debugplugin-js/src/DebugPlugin.ts:183]: remove

File: extensions/applicationinsights-debugplugin-js/src/DebugPlugin.ts [hunk 5]
```diff
@@ -152,56 +152,35 @@ export default class DebugPlugin extends BaseTelemetryPlugin {
...
-                protoFns.push(key);
+          for (let i = 0; i < trackers.length; i++) {
+            let tracker = trackers[i];
+            for (const target of [propertiesExt, analyticsExt, ajaxDependencyExt, channelPluginExt, diagLog]) {
+              let val = InstrumentFunc(target, tracker, {
+                req: _self.preProcessItem(tracker) as any as () => InstrumentorHooksCallback,
+                rsp: _self.postProcessItem(tracker) as any as () => InstrumentorHooksCallback
+              }, true);
+
+              if (!val) {
```
Comment (MSNev) [inline extensions/applicationinsights-debugplugin-js/src/DebugPlugin.ts:167]: As your passing true to the checkPrototype argument (last one) above you this should be dead code as the above will already check and hook the prototype.

File: extensions/applicationinsights-debugplugin-js/src/DebugPlugin.ts (hunk 3 above)
Comment (MSNev) [reply extensions/applicationinsights-debugplugin-js/src/DebugPlugin.ts:159]: Or better use the arrForEach() helper

File: extensions/applicationinsights-debugplugin-js/src/DebugPlugin.ts (hunk 4 above)
Comment (MSNev) [reply extensions/applicationinsights-debugplugin-js/src/DebugPlugin.ts:183]: +1 - The console is actually not defined by some browsers unless the dev console is open.

**Pull request merged**